frontend/packages/console-app/src/components/network-policies/network-policy-form.tsx

```diff
@@ -35,56 +39,67 @@ const emptyRule = (): NetworkPolicyRule => {
 };
 
 type NetworkPolicyFormProps = {
-  namespace: string;
+  formData: NetworkPolicyKind;
+  onChange: (newFormData: NetworkPolicyKind) => void;
 };
 
 export const NetworkPolicyForm: React.FunctionComponent<NetworkPolicyFormProps> = (props) => {
   const { t } = useTranslation();
-  const { namespace } = props;
+  const { formData, onChange } = props;
 
-  const emptyPolicy: NetworkPolicy = {
-    name: '',
-    namespace,
-    podSelector: [['', '']],
-    ingress: {
-      denyAll: false,
-      rules: [],
-    },
-    egress: {
-      denyAll: false,
-      rules: [],
-    },
-  };
+  const normalizedK8S = networkPolicyNormalizeK8sResource(formData);
+  const converted = networkPolicyFromK8sResource(normalizedK8S, t);
+  const [networkPolicy, setNetworkPolicy] = React.useState(converted);
 
-  const [networkPolicy, setNetworkPolicy] = React.useState(emptyPolicy);
   const [inProgress, setInProgress] = React.useState(false);
   const [error, setError] = React.useState('');
   const [showSDNAlert, setShowSDNAlert] = React.useState(true);
 
+  if (isNetworkPolicyConversionError(networkPolicy)) {
+    // Note, this case is not expected to happen. Validity of the network policy for form should have been checked prior to showing this form.
```

Comment:
I'm unclear on how we handle this. Do we disable the radio button to switch back to the form if the YAML is invalid?

Reply:
It's actually something already implemented in the `SyncedEditor` : when the `onChange` func props throws an error, it will display a message like this:

![Capture d’écran de 2021-09-13 10-52-59](https://user-images.githubusercontent.com/2153442/133054793-5aaf5323-7159-4055-9308-df4fde9c9756.png)

Hence preventing user from switching to the form view unless changes are discarded, hence ensure the form view always received correct data.

Reply:
I've added a second comment line to better explain why it shouldn't run into this conditional block

Reply:
@jotak Can we tell the user specifically what's wrong? Or they won't know how to fix it.

Reply:
That would be something to change in the SyncedEditor (currently the content of the error is silently discarded, cf the catch block https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/components/synced-editor/index.tsx#L62 ). I guess we could display the error as part of the message, however we'd better review all the current usages and make sure all the errors are ok to display (I know it's also used in OLM, at least)

Reply:
would it be ok to create a new task for that?

Reply:
Can this controlled by a prop on `SyncedEditor`? Then we can opt in per form until we can make sure all the messages are good.

Adding @christianvogt and @rohitkrai03 since `SyncedEditor` seems to primarily be used in the developer perspective.

Reply:
cc @TheRealJon as well since we used `SyncedEditor` for the operand form.

Reply:
@spadgett done: controlled by a prop, opt-in. Here's how it's like in network policies :

![Capture d’écran de 2021-09-15 14-32-56](https://user-images.githubusercontent.com/2153442/133434747-debd1839-7717-4774-a55f-95fb599d5088.png)

Note, I haven't tested the olm / operand form view as I haven't found how to view it :/